megamek/src/megamek/client/ui/swing/MovementDisplay.java

```diff
@@ -2646,7 +2646,7 @@ private synchronized void updateLoadButtons() {
         // we will say that it is not possible to unload units in such a situation.
         if (finalCoordinatesOnBoard) {
             for (Entity en : loadedUnits) {
-                if (en.isElevationValid(unloadEl, hex) || (en.getJumpMP() > 0)) {
+                if (ce.isSupportVehicle() || en.isElevationValid(unloadEl, hex) || (en.getJumpMP() > 0)) {
```

Comment:
Not sure about this logic - why would we ignore the valid elevation/jump MP check for support vehicles?

Reply:
Hmmm you caught me there. It kinda works because this only activates the button while the legal unload hexes are determined elsewhere so as the worst result the list of unload positions shown to the player would be empty. Not exactly an improvement though. The elevation check is invalid for SV unload as it only checks the hex of the transport. I'll have to take a closer look. This method looks like a patchwork of stuff to me.